### PR TITLE
Add info about FastTESR and some other improvements

### PR DESCRIPTION
**Pull request by diesieben07**

### CODE REVIEW

**Review by elifoster — DISMISSED** [review]
(I'm on mobile so i can't put a line comment.)

The very last sentence is a comma splice; the comma should be changed to a period:

File: docs/tileentities/tesr.md [hunk 1]
```diff
@@ -15,12 +17,19 @@ Only one TESR exists for a given tile entity. Therefore, values that are specifi
...
 ------------------
 
 In order to register a TESR, call `ClientRegistry#bindTileEntitySpecialRenderer` passing the tile entity class to be renderer with this TESR and the instance of the TESR to use to render all TEs of this class.
+
+`FastTESR`
+----------
+
+A TESR can opt-in to being a FastTESR by extending the `FastTESR` class instead of `TileEntitySpecialRenderer`. Instead of implementing `renderTileEntityAt`, `renderTileEntityFast` must be implemented.
+
+A FastTESR can offer performance improvements over a traditional TESR and should be used wherever possible. This is due to the fact that all FastTESR instances are batched together and only issue _one_ combined draw call for all FastTESRs per frame to the GPU. This advantage comes at the cost of making direct OpenGL access via `GlStateManager` or the `GLXX` classes impossible. Instead a FastTESR must only add vertices to the provided `VertexBuffer`, which represents the combined vertex data for all FastTESRs. This allows rendering `IBakedModel`s, an example can be found in Forge's `AnimationTESR`.
```
Comment (williewillus) [inline docs/tileentities/tesr.md]: you also need to override `TileEntity.hasFastRenderer` to true

**Comment by diesieben07** [comment]
> The very last sentence is a comma splice; the comma should be changed to a period
No, it is not. It's an example for rendering IBakedModels, which is not the only usecase for FastTESR.

**Comment by elifoster** [comment]
@diesieben07 I'm talking about grammar. `This allows rendering IBakedModels, an example can be found in Forge's AnimationTESR.` is two independent clauses.

**Comment by diesieben07** [comment]
I know what you are talking about. I am not a native English speaker, but this comma seems correct to me. Because it is _not_ two independent clauses.
Think of "This allows rendering IBakedModels, an example _of which_ can be found in ...".

**Comment by elifoster** [comment]
If the example is specifically for rendering IBakedModels, it should be joined with a conjunction and a comma (in which case I'll fix my fix). However to me, with it as a splice or not, it sounds like it is simply providing an example of FastTESRs in general.

**Comment by diesieben07** [comment]
It kind of is. Whatever, it doesn't matter. Just "This allows Rendering IBakedModels." as it's own sentence sounds stupid imho.

**Pull request merged**